src/client/content_cao.cpp

```diff
@@ -1131,6 +1135,18 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 		rot_translator.val_current = m_rotation;
 		updateNodePos();
 	}
+
+	if (m_animated_meshnode) {
+		// Everything must be updated; the whole transform
+		// chain as well as the animated mesh node.
+		// Otherwise, bone attachments would be relative to
+		// a position that's one frame old.
+		if (m_matrixnode)
+			updatePositionRecursive(m_matrixnode);
+		m_animated_meshnode->updateAbsolutePosition();
```

Comment:
Is this necessary? The absolute position is updated in `updateBonePosition` -> `updateAbsolutePositionOfAllChildren`

https://github.com/zaki/irrlicht/blob/master/source/Irrlicht/CBoneSceneNode.cpp#L92

Reply:
It's called on the root bone, not this scene node itself. This is one level above. (and must be updated first)

Reply:
Thanks. That explains it.